### PR TITLE
Add context-aware task submission

### LINKED ISSUE #43: Add context-aware task submission
## Background

`Submit` currently has no context-aware cancellation path. In BLOCK mode, a submission can wait for queue space without a way to stop waiting, and a queued task cannot be skipped when the caller context is already canceled before execution. Also, nil tasks can affect `Wait()` bookkeeping if accepted.

## Proposal

- Add `SubmitCtx(ctx, task)` for context-aware submission.
- Let cancellation stop a BLOCK-mode submission while it is waiting for queue space.
- Skip queued tasks when their context is canceled before a worker starts them.
- Keep running-task cancellation cooperative: task closures should check `ctx.Done()` or pass `ctx` downstream.
- Ignore nil tasks in `Submit` and `SubmitBefore` so invalid submissions do not affect `Wait()`.
- Update English and Chinese README docs and add regression tests.

## Validation

The planned tests cover nil tasks, cancellation before submission, cancellation while queued, cooperative cancellation while running, and cancellation while waiting for queue space.

### PR DESCRIPTION
**Pull request by amigoer**
## Summary

- add `SubmitCtx(ctx, task)` for context-aware submission and queue waiting cancellation
- skip queued context tasks when the context is canceled before execution
- ignore nil tasks in `Submit` and `SubmitBefore` so invalid submissions do not affect `Wait()`
- document `SubmitCtx` behavior in English and Chinese README files
- add regression tests for nil tasks and context cancellation states

Closes #43

## Tests

- `go test -run 'TestAgilePoolSubmit.*Nil|TestAgilePoolSubmitCtx' ./...`
- `go test -race -run 'TestAgilePoolSubmit.*Nil|TestAgilePoolSubmitCtx' ./...`
- `go vet ./...`
- `git diff --check`
- `go test -short ./...`

### CODE REVIEW

File: pool.go [hunk 1]
```diff
@@ -112,6 +112,31 @@ func (p *Pool) SetLogger(l Logger) {
...
+		return
+	}
+
+	p.submit(&contextTask{
+		ctx:  ctx,
+		task: task,
+	}, ctx)
+}
+
+func (p *Pool) submit(task Task, ctx context.Context) {
```
Comment (jadeproheshan) [inline pool.go]: 修改：我打算做个小改动，调一下这里的参数顺序，一是跟导出函数的顺序对齐，二是context一般放前面

File: pool.go [hunk 2]
```diff
@@ -112,6 +112,31 @@ func (p *Pool) SetLogger(l Logger) {
 }
 
 func (p *Pool) Submit(task Task) {
+	if isNilTask(task) {
+		return
+	}
+
+	p.submit(task, nil)
```
Comment (jadeproheshan) [inline pool.go]: 询问：这里改为p.submit(context.Background(), task)怎么样

然后后面worker满后的队列逻辑就可以统一为
select {
  case p.taskQueue <- task:
  case <-ctx.Done():
        p.wg.Done()
        return
  }
因为backgrout context会返回空channel，所以第二个case就不会被执行

**Review by jadeproheshan — APPROVED** [review]

File: pool.go (hunk 2 above)
Comment (amigoer) [reply pool.go]: 可以的，这个我认为没问题，也更健壮了

File: pool.go (hunk 1 above)
Comment (amigoer) [reply pool.go]: 可以，没问题，标准做法第一个参数就是 context

**Review by Yiming1997 — APPROVED** [review]
